File: src/main/java/gg/obsidian/discordbridge/DiscordConnection.java
package gg.obsidian.discordbridge;

import me.itsghost.jdiscord.DiscordAPI;
import me.itsghost.jdiscord.DiscordBuilder;
import me.itsghost.jdiscord.Server;
import me.itsghost.jdiscord.talkable.Group;

public class DiscordConnection implements Runnable {

    public final DiscordBridge plugin;
    public DiscordAPI api;
    public Server server;
    public Group channel;

    public DiscordConnection(DiscordBridge plugin) {
        this.plugin = plugin;
    }

    public void run() {
        try {
            api = new DiscordBuilder(plugin.email, plugin.password).build().login();
            api.getEventManager().registerListener(new DiscordListener(plugin, api));
        } catch (Exception e) {
            plugin.getLogger().severe("Error connecting to Discord: " + e);
        }
    }

    public void send(String name, String message) {
        if (server == null && (server = getServerById(plugin.serverID)) == null) {
            return;
        }

        if (channel == null && (channel = getGroupByName(server, plugin.channel)) == null) {
            return;
        }

        channel.sendMessage("<" + name + "> " + message);
    }

    private Server getServerById(String id) {
        for (Server server : api.getAvailableServers())
            if (server.getId().equals(id))
                return server;
        return null;
    }

    private Group getGroupByName(Server server, String name) {
        for (Group group : server.getGroups())
            if (group.getName().equals(name))
                return group;
        return null;
    }
}

File: src/main/scala/gg/obsidian/discordbridge/DiscordConnection.scala
package gg.obsidian.discordbridge

import me.itsghost.jdiscord.talkable.Group
import me.itsghost.jdiscord.{DiscordAPI, DiscordBuilder, Server}

import scala.collection.JavaConversions._

class DiscordConnection(plugin: DiscordBridge) extends Runnable {

  var api: DiscordAPI = _
  var server: Option[Server] = _
  var channel: Option[Group] = _

  def run() = {
    try {
      api = new DiscordBuilder(plugin.email, plugin.password).build().login()
      api.getEventManager().registerListener(new DiscordListener(plugin, api))
    } catch {
      case e: Exception => plugin.getLogger().severe("Error connecting to Discord: " + e)
    }
  }

  def send(name: String, message: String): Unit = {
    server = server match {
      case None => getServerById(plugin.serverID)
      case _ => server
    }

    if (server.isEmpty) return

    channel = channel match {
      case None => getGroupByName(server.get, plugin.channel)
      case _ => channel
    }

    if (channel.isEmpty) return

    channel.get.sendMessage("<" + name + "> " + message)
  }

  def getServerById(id: String): Option[Server] = {
    api.getAvailableServers.toList.foreach { server: Server => {
      if (server.getId().equals(id)) return Some(server)
    }}
    None
  }

  def getGroupByName(server: Server, name: String): Option[Group] = {
    server.getGroups().toList.foreach { group: Group => {
      if (group.getName().equals(name)) return Some(group)
    }}
    None
  }
}
